hoodie-client/src/main/java/com/uber/hoodie/io/HoodieAppendHandle.java

```diff
@@ -128,8 +134,9 @@ private void init(Iterator<HoodieRecord<T>> recordItr) {
       }
       // update the new location of the record, so we know where to find it next
       record.setNewLocation(new HoodieRecordLocation(commitTime, fileId));
-    });
-    this.records = records;
+      averageRecordSize = SizeEstimator.estimate(record);
```

Comment:
have you measured how close this estimate is to reality? i.e are you able to achieve data blocks of that size?

Reply:
Did some basic testing, need to do some more.

Reply:
what were the results? I am wondering if we should estimate or use obtain the length directly from serialized values..